src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAudienceRestrictionCondition.cs

```diff
@@ -34,6 +34,20 @@ public SamlAudienceRestrictionCondition(IEnumerable<Uri> audiences)
             Audiences = (audiences == null) ? throw LogArgumentNullException(nameof(audiences)) : new List<Uri>(audiences);
         }
 
+        /// <summary>
+        /// Creates an instance of <see cref="SamlAudienceRestrictionCondition"/>.
+        /// </summary>
+        /// <param name="audiences">An <see cref="IEnumerable{String}"/> containing the audiences for a <see cref="SamlAssertion"/>.</param>
+        internal SamlAudienceRestrictionCondition(IEnumerable<string> audiences)
+        {
+            if (audiences == null)
+                throw LogArgumentNullException(nameof(audiences));
+
+            List<Uri> audienceUris = new();
+            foreach (var aud in audiences) { audienceUris.Add(new Uri(aud)); }
```

Comment:
we generally dont follow this style

Reply:
Which style is that?

Reply:
What happens is aud is not a Uri?

Reply:
Good point, I imagine we should throw an exception? Or should we return to the previous state where we only accept Uri objects and this cast should be made outside of this class?